server/src/main/scala/za/co/absa/atum/server/api/repository/PartitioningRepositoryImpl.scala

```diff
@@ -39,13 +45,30 @@ class PartitioningRepositoryImpl(
     dbCallWithStatus(createOrUpdateAdditionalDataFn(additionalData), "createOrUpdateAdditionalData")
   }
 
+  override def getPartitioningMeasures(partitioning: PartitioningSubmitDTO):
+    IO[DatabaseError, Seq[MeasureDTO]] = {
+    getPartitioningMeasuresFn(partitioning)
+      .mapError(err => DatabaseError(err.getMessage))
+  }
+
+  override def getPartitioningAdditionalData(partitioning: PartitioningSubmitDTO):
+    IO[DatabaseError, AdditionalDataDTO] = {
+    getPartitioningAdditionalDataFn(partitioning).mapBoth(err => DatabaseError(err.getMessage), _.toMap)
+  }
 }
 
 object PartitioningRepositoryImpl {
-  val layer: URLayer[CreatePartitioningIfNotExists with CreateOrUpdateAdditionalData, PartitioningRepository] = ZLayer {
+  val layer: URLayer[CreatePartitioningIfNotExists with GetPartitioningMeasures with
+    GetPartitioningAdditionalData with CreateOrUpdateAdditionalData, PartitioningRepository] = ZLayer {
```

Comment:
consider better formatting, like

```suggestion
   val layer: URLayer[
    CreatePartitioningIfNotExists
      with GetPartitioningMeasures
      with GetPartitioningAdditionalData
      with CreateOrUpdateAdditionalData,
    PartitioningRepository
  ] = ZLayer {
```